src/sonic-daemon-base/sonic_daemon_base/daemon_base.py

```diff
@@ -38,6 +37,7 @@
 #
 
 def db_connect(db):
+    from swsscommon import swsscommon
```

Comment:
what is the benefit of this change?

Reply:
Hi, 
for the following reasions:
1. sonic_platform is loaded on the host where swsscomm isn't installed. 
2. swsscommon is only required when db_connect is called.
3. db_connect won't be called on host.
by calling "import swsscommon" only when db_connect being called, we can remove the dependency for swsscommon on host.

Reply:
@stephenxs a few comments here:

1. sonic_platform is loaded on the host where swsscomm isn't installed.
Agree
2. swsscommon is only required when db_connect is called.
Agree
3. db_connect won't be called on host.
Disagree. If someone use it - it will crash

IMHO, this looks like a WA. We need more robust solution here.

Reply:
Would it be better if we installed swsscommon on the host? I can see it being needed in the future, regardless.

Reply:
> Would it be better if we installed swsscommon on the host? I can see it being needed in the future, regardless.

If it is required in the feature, I'll change the logic to install it on the host.

Reply:
> @stephenxs a few comments here:
> 
> 1. sonic_platform is loaded on the host where swsscomm isn't installed.
>    Agree
> 2. swsscommon is only required when db_connect is called.
>    Agree
> 3. db_connect won't be called on host.
>    Disagree. If someone use it - it will crash
> 
> IMHO, this looks like a WA. We need more robust solution here.

Yes. Joe suggests to install it on the host. I'll do it in this way.

Reply:
I'm not suggesting you install it here; I'm just wondering if we should install it by default in the host image.

Reply:
got it.
let's list the pros and cons of installing it on the host: 
pros:
1. less confusing. 
    to install a package on the host without all of its functions callable is confusing.
cons:
1. install a couple of packages which are not currently required

BTW I've just tried installing it on the host and found it need many other dependencies which are not installed. 
```
admin@mtbc-sonic-01-2410:~$ sudo dpkg -i libswsscommon_1.0.0_amd64.deb 
Selecting previously unselected package libswsscommon.
(Reading database ... 26155 files and directories currently installed.)
Preparing to unpack libswsscommon_1.0.0_amd64.deb ...
Unpacking libswsscommon (1.0.0) ...
dpkg: dependency problems prevent configuration of libswsscommon:
 libswsscommon depends on libhiredis0.14 (>= 0.14.0); however:
  Package libhiredis0.14 is not installed.
 libswsscommon depends on libnl-3-200 (>= 3.2.21); however:
  Package libnl-3-200 is not installed.
 libswsscommon depends on libnl-genl-3-200 (>= 3.2.7); however:
  Package libnl-genl-3-200 is not installed.
 libswsscommon depends on libnl-route-3-200 (>= 3.2.7); however:
  Package libnl-route-3-200 is not installed.
```
In this sense, I think it's better to raise another PR to install it on the host if we need to do so.

Reply:
Thanks for the info, Stephen. I agree it's better to raise another PR to install it on the host if we need to do so.